src/ess/reflectometry/tools.py

```diff
@@ -270,3 +271,25 @@ def combine_curves(
         ),
         coords={'Q': qgrid},
     )
+    if any('Q_resolution' in c.coords for c in curves):
+        # This might need to be revisited. The question about how to combine curves
+        # with different Q-resolution is not completely resolved.
+        # However, in practice the difference in Q-resolution between different curves
+        # is small so it's not likely to make a big difference.
+        q_res = (
+            sc.DataArray(
+                data=c.coords.get(
+                    'Q_resolution', sc.scalar(float('nan')) * sc.values(c.data.copy())
```

Comment:
Odd default. Is that really the unit you want? How about something like `sc.full_like(c.coords['Q'], value=nan)`?